ground/src/main/java/com/google/android/ground/system/LocationManager.kt

```diff
@@ -79,6 +86,40 @@ constructor(
       .onErrorReturn { Result.failure(it) }
   }
 
+  fun toggleLocationLock() {
+    if (locationLockState.value.getOrDefault(false)) {
+      unlock()
```

Comment:
Nit: `disableLocationLock()` might be more appropriate, since we've been talking about "location lock" as a feature that can be turned on and off, not about "locking the location".